src/azul/tdr.py

```diff
@@ -0,0 +1,45 @@
+from functools import cached_property
+from typing import (
+    List,
+    NamedTuple,
+)
+
+from azul.bigquery import (
+    BigQueryAdaptee,
+    GoogleQueryAdapter,
+)
+from azul.indexer import BundleFQID
+from azul.uuids import validate_uuid_prefix
+
+
+class BigQueryDataset(NamedTuple):
+    id: str
+    name: str
+    is_snapshot: bool
```

Comment:
Looking at the usages of this attribute, there seems to be no enforcement of the uniqueness constraint in snapshots. Like any other table, the links table in a TDR dataset can contain multiple versions of the same subgraph. Should that not be asserted or dealt with somehow?